File: templates/python-agent/src/agent_runtime/planner.py
from dataclasses import dataclass, field
from typing import Protocol
from .contracts import PlanStep

@dataclass
class AgentPlan:
    goal: str
    steps: list[PlanStep]
    completion_criteria: list[str]
    assumptions: list[str] = field(default_factory=list)
    risks: list[str] = field(default_factory=list)

class Planner(Protocol):
    def create(self, objective: str) -> AgentPlan: ...
    def revise(self, plan: AgentPlan, evidence: list[str]) -> AgentPlan: ...

File: templates/typescript-agent/src/runtime/planner.ts
import { PlanStep } from "../contracts.js";
export interface AgentPlan { goal: string; assumptions: string[]; risks: string[]; requiredCapabilities: string[]; steps: PlanStep[]; completionCriteria: string[]; }
export interface Planner { create(objective: string): Promise<AgentPlan>; revise(plan: AgentPlan, evidence: string[]): Promise<AgentPlan>; }
